### PR TITLE
Add timing on the client even if no behavior is configured on the service side

### PR DESCRIPTION
**Pull request by sarustamyan-bdb**

### CODE REVIEW

File: StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs [hunk 1]
```diff
@@ -121,6 +123,8 @@ public void AfterReceiveReply(ref Message reply, object correlationState)
 
                     profiler.AddProfilerResults(resultsHeader.ProfilerResults);
                 }
+
+                profiler.Head?.Stop();
```
Comment (hadason) [inline StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs]: Why are you relying on `Head`, you can pass the Step as part of the `CorrelationState` and use it. I think it's a safer way.

**Review by hadason — CHANGES_REQUESTED** [review]

File: StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs (hunk 1 above)
Comment (sarustamyan-bdb) [reply StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs]: Fixed

File: StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs [hunk 2]
```diff
@@ -84,53 +78,64 @@ public object BeforeSendRequest(ref Message request, IClientChannel channel)
...
                 }
             }
+            else
+            {
+                throw new InvalidOperationException("MVC Mini Profiler does not support EnvelopeNone unless HTTP is the transport mechanism");
+            }
+
+            if (resultsHeader == null || resultsHeader.ProfilerResults == null)
+            {
+                return;
```
Comment (hadason) [inline StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs]: Why are you returning here without stopping the profiler? isn't the whole idea to add our own timing of there's no explicit timing in the WCF result?

**Review by hadason — CHANGES_REQUESTED** [review]

File: StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs (hunk 2 above)
Comment (sarustamyan-bdb) [reply StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs]: You're right, my bad. Fixed

**Review by hadason — APPROVED** [review]

**Pull request merged**